source/services/viber.js

```diff
@@ -0,0 +1,19 @@
+/**
+ * Viber service provider
+ */
+
+export default {
+    popupUrl: 'viber://forward?text={content}',
+    click() {
+        if (Object.prototype.hasOwnProperty.call(this.widget.dataset, 'comment')) {
```

Comment:
What's the difference with `this.widget.dataset.hasOwnProperty('comment')`?

Reply:
https://stackoverflow.com/questions/12017693/why-use-object-prototype-hasownproperty-callmyobj-prop-instead-of-myobj-hasow :)

Reply:
@NikolayRys and why do we need it here exactly? I see two reasons described in the article:

- an object might not have the prototype chain (Object.create(null)) but it's not our case, we're creating the object using literal `{}` right here, aren't we?
- an object might have hasOwnProperty function redefined but it's not our case as well, it's just a plain object right in from of our eyes

So, why do _we_ need it here?

Reply:
If we consider it as a good practice, I agree with that. That's fine. In general, I'm against the way of doing things like this since it becomes a cargo-cult eventually.

Reply:
Thanks for sharing the link, btw. I didn't know about those problems 👍

Reply:
No worries :)
well, in any case, the project linter also enforces this :)
If we don't want it(and I would be okay with this), then we need to turn it off.

Reply:
> the project linter also enforces this 

Yeah, didn't know that. If it's a project rule then it's fine, for sure.